pkg/resource/generator.go

```diff
@@ -86,8 +96,132 @@ func (g *Generator) list(cr *regopapi.ImageRegistry) ([]Mutator, error) {
 	return mutators, nil
 }
 
+// syncStorage checks:
+// 1.)  to make sure that an existing storage medium still exists and we can access it
+// 2.)  to see if the storage medium name changed and we need to:
+//      a.) check to make sure that we can access the storage or
+//      b.) see if we need to try to create the new storage
+func (g *Generator) syncStorage(cr *regopapi.ImageRegistry, modified *bool) error {
+
+	// If the Image Registry is currently using S3 storage
+	if cr.Spec.Storage.S3 != nil {
+		// Create a new driver with the current configuration
+		// This saves us from having to duplicate code
+		driver := storageS3.NewDriver(cr.Name, cr.Namespace, cr.Spec.Storage.S3)
+
+		// If the bucket name has changed
+		if cr.Status.Storage.State.S3 != nil && cr.Spec.Storage.S3.Bucket != cr.Status.Storage.State.S3.Bucket {
+			driver.Config.Bucket = cr.Spec.Storage.S3.Bucket
+			// Check to see if the bucket exists
+			// and if we can access it
+			if err := driver.CheckBucketExists(cr); err != nil {
+				installConfig, err := clusterconfig.GetInstallConfig()
+				if err != nil {
+					return err
+				}
+				// If the bucket doesn't exist, try to create it
+				if err := driver.CreateAndTagBucket(installConfig, cr); err != nil {
```

Comment:
if we create the bucket (after a name change, or basically ever) that has implications for the ManagedStorage value, so it needs to be updated, right?

(Suppose we started out using a bucket they created, so ManagedStorage=false, we won't delete the bucket on teardown...  then they change the bucketname to something else and we *do* create the bucket...now we own it.  If they change the bucket name to something else that already exists, we don't own it.  So you need to be keeping that state updated)

Reply:
This should be updated now.